src/ui/public/vis/editors/default/default.js

```diff
@@ -64,19 +64,24 @@ const defaultEditor = function ($rootScope, $compile) {
 
           updateScope();
 
+          $scope.state = $scope.vis.copyCurrentState();
+          $scope.oldState = $scope.vis.getSerializableState($scope.state);
+
           $scope.toggleSidebar = () => {
             $scope.$broadcast('render');
           };
 
           this.el.one('renderComplete', resolve);
-
           // track state of editable vis vs. "actual" vis
           $scope.stageEditableVis = () => {
+            $scope.vis.setCurrentState($scope.state);
+            $scope.oldState = $scope.vis.getSerializableState($scope.state);
             $scope.vis.updateState();
             $scope.vis.dirty = false;
           };
           $scope.resetEditableVis = () => {
-            $scope.vis.resetState();
```

Comment:
With removal of `resetState()` here, there is only one internal usage left in `vis.setState`, which could be perfectly inlined to:

```js
setState(state, updateCurrentState = true) {
  this._state = _.cloneDeep(state);
  if (updateCurrentState) {
    this.setCurrentState(this._state);
  }
}
```

That way we could remove the `resetState` completely, which imho doesn't make too much sense anymore, since it was used for resetting the editor only afaik.